core/flyout_vertical.js

```diff
@@ -323,15 +323,23 @@ Blockly.VerticalFlyout.prototype.getClientRect = function() {
     // https://bugzilla.mozilla.org/show_bug.cgi?id=1066435
     if (goog.userAgent.GECKO &&
         this.targetWorkspace_ && this.targetWorkspace_.isMutator) {
-      // If we're in a mutator, its scale is always 1, purely because of some
-      // oddities in our rendering optimizations.  The actual scale is the same
-      // as the scale on the parent workspace.
-      var scale = this.targetWorkspace_.options.parentWorkspace.scale;
-      var altX = x + this.leftEdge_ * scale;
-      // If the browser was obviously wrong, use the calculated value.  If the
-      // browser's value was plausible, trust it.
-      if (Math.abs(altX - x) > 10) {
-        x = altX;
+      // The position of the left side of the mutator workspace in pixels
+      // relative to the window origin.
+      var targetWsLeftPixels =
+          this.targetWorkspace_.svgGroup_.getBoundingClientRect().x;
+      // The client rect is in pixels relative to the window origin.  When the
+      // browser gets the wrong value it reports that the flyout left is the
+      // same as the mutator workspace left.
+      // We know that in a mutator workspace with the flyout on the right, the
+      // visible area of the workspace should be more than ten pixels wide.  If
+      // the browser reports that the flyout is within ten pixels of the left
+      // side of the workspace, ignore it and manually calculate the value.
+      if (Math.abs(targetWsLeftPixels - x) < 10) {
```

Comment:
Will the 10px assumption be invalidated if we allow mutators without flyouts/with empty flyouts, à la #1759. In practice, a mutator with an empty/invisible empty toolbox (no way to add new blocks), probably doesn't a need for a block delete area.  Technically, the issues are distinct.

Reply:
At that point there should be no flyout, so nothing to call `getClientRect` on.  Then this code won't be hit.